Makefile

```diff
@@ -1,59 +1,111 @@
+GHC_FLAGS = --ghc-options "$(FLAGS)"
+endif
 
-# Build the executables with nix (also builds the test suite).
-nix-build-executables:
-	@ nix build .#check.${current-system}
+nix_build:
+	nix build -L .#check.x86_64-linux
 
-# Build the tests with nix.
-nix-build-test:
-	@ nix build .#seabug-onchain:test:seabug-onchain-tests
+build: requires_nix_shell
+	cabal v2-build $(GHC_FLAGS)
 
-# Starts a ghci repl inside the nix environment.
-nix-cabal-repl:
-	@ nix develop -c cabal new-repl
+watch: requires_nix_shell
+	while sleep 1; do find src seabug-onchain.cabal | entr -cd make build; done
 
-# Target to use as dependency to fail if not inside nix-shell.
-requires_nix_shell:
-	@ [ "($IN_NIX_SHELL)" ] || echo "The $(MAKECMDGOALS) target must be run from inside `nix develop`"
-	@ [ "($IN_NIX_SHELL)" ] || (echo "    run `nix develop` first" && false)
+test: requires_nix_shell
+	cabal v2-test
+
+ghci: requires_nix_shell
+	cabal v2-repl $(GHC_FLAGS) seabug-onchain
+
+# Source dirs to run fourmolu on
+FORMAT_SOURCES := $(shell fd -e hs)
+
+# Extensions we need to tell fourmolu about
+FORMAT_EXTENSIONS := -o -XTemplateHaskell -o -XTypeApplications -o -XImportQualifiedPost -o -XPatternSynonyms -o -fplugin=RecordDotPreprocessor
+
+# Run fourmolu formatter
+format: requires_nix_shell
+	fourmolu --mode inplace --check-idempotence $(FORMAT_EXTENSIONS) $(FORMAT_SOURCES)
+
+# Check formatting (without making changes)
+format_check: requires_nix_shell
+	fourmolu --mode check --check-idempotence $(FORMAT_EXTENSIONS) $(FORMAT_SOURCES)
 
-# Build with Stack - independent of NIX.
-stack-build: 
-	stack build --ghc-options="-Wall"
+# Cabal package definitions
+CABAL_SOURCES := $(shell fd -e cabal)
 
-# Test with Stack.
-stack-test: 
-	stack test all 
+cabalfmt: requires_nix_shell
+	cabal-fmt --inplace $(CABAL_SOURCES)
 
-# Watch with Stack.
-stack-watch: 
-	stack build --file-watch --ghc-options="-Wall"
+cabalfmt_check: requires_nix_shell
+	cabal-fmt --check $(CABAL_SOURCES)
 
-# Watch Test with Stack.
-stack-test-watch: 
-	stack test --file-watch 
+# Nix files to format
+NIX_SOURCES := $(shell fd -e nix)
+
+nixpkgsfmt: requires_nix_shell
+	nixpkgs-fmt $(NIX_SOURCES)
+
+nixpkgsfmt_check: requires_nix_shell
+	nixpkgs-fmt --check $(NIX_SOURCES)
+
+# Check with hlint, currently I couldn't get --refactor to work
+lint: requires_nix_shell
+	hlint $(FORMAT_SOURCES)
+
+readme_contents:
+	echo "this command is not nix-ified, you may receive an error from npx"
+	npx markdown-toc ./README.md --no-firsth1
+
+# Target to use as dependency to fail if not inside nix-shell
+requires_nix_shell:
+	@ [ "$(IN_NIX_SHELL)" ] || echo "The $(MAKECMDGOALS) target must be run from inside nix-shell"
+	@ [ "$(IN_NIX_SHELL)" ] || (echo "    run 'nix-shell --pure' first" && false)
 
-# Add folder locations to the list to be reformatted.
-fourmolu-format:
-	@ echo "> Formatting all .hs files"
-	./format.sh
+build_path = dist-newstyle/build/x86_64-linux/ghc-8.10.4.20210212/seabug-onchain-0.1
```

Comment:
I think this would need to be changed to `dist-newstyle/build/x86_64-linux/ghc-8.10.7/seabug-onchain-0.1.0.0/` for `clear_build` to do anything